addons/sourcemod/scripting/freak_fortress_2.sp

```diff
@@ -6543,182 +6553,248 @@ stock GetBossIndex(client)
+				LogError("[FF2] %s's %s formula contains invalid character %c!", bossName, key, buffer[0]);
+				break;
 			}
 		}
-
-		if(valueReady)
+		if (foundOperator || closeBracket)
 		{
-			valueReady=false;
-			if(usePlayers)
+			if (foundOperator)
 			{
-				usePlayers=false;
-				switch(_operator[parentheses])
+				foundOperator = false; //This here is just so I don't forget to tell the code, "We already operated on."
+			}
+			if (closeBracket)
+			{
+				closeBracket = false;
+				bracket--; //We need to operate on the 1 less bracket
+				switch(_operator[bracket])
 				{
-					case 1:
+					case Operation_Add:
 					{
-						sum[parentheses]+=playing;
+						sum[bracket] += sum[bracket+1];
+						break;
 					}
-					case 2:
+					case Operation_Subtract:
 					{
-						sum[parentheses]-=playing;
+						sum[bracket] -= sum[bracket+1];
+						break;
 					}
-					case 3:
+					case Operation_Multiplication:
 					{
-						sum[parentheses]*=playing;
+						sum[bracket] *= sum[bracket+1];
+						break;
 					}
-					case 4:
+					case Operation_Division:
 					{
-						sum[parentheses]/=playing;
+						sum[bracket] /= sum[bracket+1];
+						break;
 					}
-					case 5:
+					case Operation_Exponential:
 					{
-						sum[parentheses]=Pow(sum[parentheses], Float:playing);
+						sum[bracket] ^= sum[bracket+1];
+						break;
 					}
 					default:
 					{
-						parentheses=1;
+						//We don't got an operation for this ending bracket.(eg:  "))")
+						sum[bracket] = sum[bracket+1];
 						break;
 					}
 				}
+				_operator[bracket+1] = Operation_None;
 			}
-
-			if(value[0]!='\0' && canAdd)
+			if (usePlayers)
 			{
-				canAdd=false;
-				switch(_operator[parentheses])
+				usePlayers = false;
+				//Add sum with players.
+				switch(_operator[bracket])
 				{
-					case 1:
+					case Operation_Add:
 					{
-						sum[parentheses]+=StringToFloat(value);
-						strcopy(value, sizeof(value), "");
+						sum[bracket] += float(playing);
+						break;
 					}
-					case 2:
+					case Operation_Subtract:
+					{
+						sum[bracket] -= float(playing);
+						break;
+					}
+					case Operation_Multiplication:
+					{
+						sum[bracket] *= float(playing);
+						break;
+					}
+					case Operation_Division:
+					{
+						sum[bracket] /= float(playing);
+						break;
+					}
+					case Operation_Exponential:
 					{
-						sum[parentheses]-=StringToFloat(value);
-						strcopy(value, sizeof(value), "");
+						sum[bracket] ^= float(playing);
+						break;
 					}
-					case 3:
+					default:
 					{
-						sum[parentheses]*=StringToFloat(value);
-						strcopy(value, sizeof(value), "");
+						sum[bracket] = float(playing);
+						break;
 					}
-					case 4:
+				}
+			}
+			//TODO: Add useLives back.
+			if (value[0] != '\0') //Value has something in it, by time we get here we already have all the value parsed(hopefully).
+			{
+				new Float:tValue = StringToFloat(value);
```

Comment:
I'd still like this to be changed-I'm sure the compiler errors can be worked out.